frontend/src/components/Customer.tsx

```diff
@@ -0,0 +1,625 @@
+    return () => clearInterval(interval); // Clear interval khi component bị unmount
+  }, [contract, currentPhase]);
+
+  // Hàm claimReward khi phase là Expiry
+  const claimReward = async () => {
+    if (contract && currentPhase === Phase.Expiry) {
+      const provider = new ethers.providers.Web3Provider(window.ethereum); // Define provider here
+      try {
+        const tx = await contract.claimReward();  // Gọi smart contract
+        await tx.wait();
+  
+        const newBalanceWei = await provider.getBalance(walletAddress);
+        const newBalanceEth = parseFloat(ethers.utils.formatEther(newBalanceWei));
+
+        const fee = (reward * 0.10); // 10% phí
+        const finalReward = reward - fee;
+        
+        setBalance(newBalanceEth);  // Cập nhật lại số dư
+        setReward(finalReward);  // Reset lại reward sau khi claim
+        setShowClaimButton(false);  // Ẩn nút claim sau khi đã nhận
+        setTotalDeposited(0); 
+        // Cập nhật lại bảng Long/Short
+        await fetchMarketDetails(); // Gọi lại hàm để cập nhật thông tin
+        await fetchContractBalance();
+
+
+        toast({
+          title: "Reward claimed!",
+          description: `You've successfully claimed your reward.`,
+          status: "success",
+          duration: 3000,
+          isClosable: true,
+        });
+      } catch (error) {
+        toast({
+          title: "Error claiming reward",
+          description: "An error occurred. Please try again.",
+          status: "error",
+          duration: 3000,
+          isClosable: true,
+        });
+      }
+    }
+  };
+  
+  
+  const canClaimReward = useCallback(async () => {
+    if (contract && currentPhase === Phase.Expiry) {
+      console.log("Checking claim eligibility..."); // Log để kiểm tra
+      try {
+        const hasClaimed = await contract.hasClaimed(walletAddress);
+        console.log("Has claimed:", hasClaimed); // Log giá trị hasClaimed
+        const oracleDetails = await contract.oracleDetails();
+        const finalPriceBN = BigNumber.from(oracleDetails.finalPrice);
+        const strikePriceBN = BigNumber.from(oracleDetails.strikePrice);
+
+        const finalPrice = parseFloat(ethers.utils.formatUnits(finalPriceBN, 0)); // Chuyển đổi giá trị cuối
+        const strikePrice = parseFloat(ethers.utils.formatUnits(strikePriceBN, 0)); // Chuyển đổi giá trị strike    
+
+        // Sửa lại việc kiểm tra `finalPrice` và `strikePrice`
+        let winningSide = finalPrice >= strikePrice ? Side.Long : Side.Short;
+
+        let userDeposit = 0;
+        if (winningSide === selectedSide) {
+          // Nếu người chơi chọn đúng bên thắng, kiểm tra khoản cược
+          userDeposit = winningSide === Side.Long ? await contract.longBids(walletAddress) : await contract.shortBids(walletAddress);
+        }
+
+        console.log("Winning side:", winningSide); // Log bên thắng
+        console.log("User deposit:", userDeposit); // Log số tiền cược của người dùng
```

Comment:
- too many codes in a single function L367:L405. separate them into functions/helper functions for better testability/readability.
- format code again if the line is too long (L405)